pkg/openfeature/api.go

```diff
@@ -0,0 +1,119 @@
+package openfeature
+
+import (
+	"errors"
+	"github.com/go-logr/logr"
+	"github.com/open-feature/go-sdk/pkg/openfeature/internal"
+	"sync"
+)
+
+// evaluationAPI wraps OpenFeature evaluation API functionalities
+type evaluationAPI struct {
+	defaultProvider FeatureProvider
+	namedProviders  map[string]FeatureProvider
+	hks             []Hook
+	evalCtx         EvaluationContext
+	logger          logr.Logger
+	mu              sync.RWMutex
+}
+
+// newEvaluationAPI is a helper to generate an API. Used internally
```

Comment:
maybe worth to augment the reasons why we don't expose the function

Reply:
The reason is that, `evaluationAPI` is the internal implementation of Flag Evaluation API. This is kept as a singleton and exposed through a public set of functions. These functions act as a facade to the implementation details. And OpenFeature SDK users do not have to be concerned about instantiation, or configuration of this. 

Here, I refactored to extract the API definition to a separate file to get some clarity. With next set of changes (ex:- init/shutdown), implementation details will be kept with API (this file & struct), with minimal/or no change to public functions.

Reply:
Thank you for the explanation. I find it weird that a file called api in a pkf openfeature doesn't expose the OF API 😅  
I know that users are supposed to utilize the client.go file, but maybe worth refactoring in a follow-up and move this into a pkg called like "internal"
e.g: [Keptn](https://github.com/keptn/keptn/tree/master/shipyard-controller/internal) and [OTel](https://github.com/open-telemetry/opentelemetry-go/tree/main/internal)

Reply:
Moving this to internal package was my original intension [1]. As you can see, this is already done for the logger

But, the dependencies of `evaluationAPI` makes this difficult as such a move will create circular dependencies. This is why I stopped the refactoring at this stage and focused on the new logic. You will see two more follow-up changes and they all benefit from this move (rather than doing everything in a single go file).

Feel free to propose a good name if you have one. Otherwise, the package refactoring should be done once spec 0.6.0 changes are done. 

[1] - https://github.com/open-feature/go-sdk/pull/180/files#diff-e679ba3761987644759cb8179818bda42002507cce86d1ec278d5f817ab09743

Reply:
oh I see the problem. Thanks again for the explanation 👍